packages/fe-fpm-writer/src/section/index.ts

```diff
@@ -113,7 +113,42 @@ function generate(
         fs.copyTpl(getTemplatePath('common/FragmentWithVBox.xml'), viewPath, completeSection);
     }
 
-    return fs;
+    return { editor: fs, section: completeSection };
+}
+
+/**
+ * Add a custom header section to an existing UI5 application.
+ *
+ * @param {string} basePath - the base path
+ * @param {CustomHeaderSection} customHeaderSection - the custom header section configuration
+ * @param {Editor} [fs] - the mem-fs editor instance
+ * @returns {Promise<Editor>} the updated mem-fs editor instance
+ */
+export function generateCustomHeaderSection(
+    basePath: string,
+    customHeaderSection: CustomHeaderSection,
+    fs?: Editor
+): Editor {
+    const manifestRoot = getManifestRoot('headersection', customHeaderSection.minUI5Version);
+    const { editor, section } = generate(basePath, customHeaderSection, manifestRoot, fs);
+    // handle fragment for templateEdit property
+    if (section.edit) {
+        const viewPath = join(section.path, `${section.edit.name}.fragment.xml`);
```

Comment:
Here code calculates `viewPath` for `templateEdit` property based on `config.edit`/`sections.edit`.
I think `section.path` is wrong and `path` should be calculated based on `config.edit`/`sections.edit`.
Here is generic code used to calculate `path` -> https://github.com/SAP/open-ux-tools/blob/main/packages/fe-fpm-writer/src/common/defaults.ts#L23

If we follow same logic for `templateEdit`, then same logic should be applied for `section.edit`, I mean something like:
```
setCommonDefaults(section.edit, manifestPath, manifest)
```